packages/http-client-csharp/emitter/src/emitter.ts

```diff
@@ -83,71 +81,87 @@ export async function $onEmit(context: EmitContext<CSharpEmitterOptions>) {
     const root = createModel(sdkContext);
 
     if (root) {
-      const generatedFolder = resolvePath(outputFolder, "src", "Generated");
+      emit(options, logger, sdkContext, context, root);
+    }
+  }
+}
 
-      if (!fs.existsSync(generatedFolder)) {
-        fs.mkdirSync(generatedFolder, { recursive: true });
-      }
+/**
+ * Emit the code model. This function is used if the sub-plugin needs to manipulate the code model.
+ * @param options - The emitter options
+ * @param logger - The logger
+ * @param sdkContext - The SDK context
+ * @param context - The emit context
+ * @param root - The code model
+ * @beta
+ */
+export async function emit(
```

Comment:
Why do we need this? We already have the createCodeModel function that can be used.

Reply:
It's used in https://github.com/Azure/azure-sdk-for-net/pull/47944/files#diff-7880b65925f80fa30f1982e8c8075ad81b357b73ef1065b636e38096113bed96
I need to manipulate the model after calling createCodeModel, and then call emit to finish the process

Reply:
Got it. createCodeModel was exposed for testing and for autorest.csharp. We don't really want MGC emitters to have to be the "driver" . Instead, we can consider exposing a callback like `updateCodeModel(codeModel: CodeModel) : CodeModel` that can be plumbed through the options. This way, emitter authors still would call OnEmit from MGC, but they don't have to handle creating the code model or writing the code model.

e.g. we would have 
``` OnEmit (Azure plugin) -> OnEmit (MGC) -> UpdateCodeModel (Azure Plugin) -> Emit code model (MGC)```